File: app/response/workspace.py
"""Plan 0023: Workspace request/response schemas."""

from __future__ import annotations

from typing import Literal, Optional
from pydantic import BaseModel, Field


class WorkspaceCreateRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    icon: Optional[str] = Field(None, max_length=50)
    color: Optional[str] = Field(None, max_length=20)


class WorkspaceUpdateRequest(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    icon: Optional[str] = Field(None, max_length=50)
    color: Optional[str] = Field(None, max_length=20)


class BindingCreateRequest(BaseModel):
    bindType: Literal["folder", "file"] = Field(...)
    folderId: Optional[int] = None
    uploadUuid: Optional[str] = None
    includeSubfolders: bool = True


class BindingResponse(BaseModel):
    id: int
    bindType: str
    folderId: Optional[int] = None
    folderName: Optional[str] = None
    uploadUuid: Optional[str] = None
    fileName: Optional[str] = None
    includeSubfolders: bool


class WorkspaceResponse(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None
    fileCount: int = 0
    chunkCount: int = 0
    bindings: list[BindingResponse] = []
    createdAt: str = ""
    updatedAt: str = ""


class WorkspaceFileItem(BaseModel):
    uploadUuid: str
    originalName: str
    mimeType: str
    vectorizable: bool = True
    vectorStatus: str = "pending"
    vectorChunkCount: int = 0

File: frontendv2/lib/api/workspace.ts
/**
 * Workspace API - Plan 0023 工作区（绑定收藏夹/文件，聚合成检索范围）.
 */
import { request } from "./client";

export interface WorkspaceBinding {
    id: number;
    bindType: "folder" | "file";
    folderId?: number;
    folderName?: string;
    uploadUuid?: string;
    fileName?: string;
    includeSubfolders: boolean;
}

export interface WorkspaceItem {
    id: number;
    name: string;
    description?: string;
    icon?: string;
    color?: string;
    fileCount: number;
    chunkCount: number;
    bindings: WorkspaceBinding[];
    createdAt: string;
    updatedAt: string;
}

export interface WorkspaceCreateParams {
    name: string;
    description?: string;
    icon?: string;
    color?: string;
}

export interface BindingCreateParams {
    bindType: "folder" | "file";
    folderId?: number;
    uploadUuid?: string;
    includeSubfolders?: boolean;
}

export const workspaceApi = {
    list: () => request<WorkspaceItem[]>("/workspaces"),
    create: (data: WorkspaceCreateParams) =>
        request<WorkspaceItem>("/workspaces", { method: "POST", body: JSON.stringify(data) }),
    get: (id: number) => request<WorkspaceItem>(`/workspaces/${id}`),
    update: (id: number, data: Partial<WorkspaceCreateParams>) =>
        request<WorkspaceItem>(`/workspaces/${id}`, { method: "PATCH", body: JSON.stringify(data) }),
    delete: (id: number) => request<{ deleted: boolean }>(`/workspaces/${id}`, { method: "DELETE" }),
    addBinding: (id: number, data: BindingCreateParams) =>
        request<WorkspaceItem>(`/workspaces/${id}/bindings`, { method: "POST", body: JSON.stringify(data) }),
    removeBinding: (workspaceId: number, bindingId: number) =>
        request<{ deleted: boolean }>(`/workspaces/${workspaceId}/bindings/${bindingId}`, { method: "DELETE" }),
    listFiles: (id: number) =>
        request<{ uploadUuid: string; originalName: string; mimeType: string; vectorizable: boolean; vectorStatus: string; vectorChunkCount: number }[]>(`/workspaces/${id}/files`),
};
